CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/layers/DayNightLayer.java

```diff
@@ -0,0 +1,327 @@
+
+    }
+
+    /**
+     * Creates at a specific location
+     *
+     * @param shadowRadius - radius of shadow
+     * @param shadowLocation - geo coordinate of shadow
+     * @param colour
+     */
+    private void createShadow(final float shadowRadius, final Location shadowLocation, final Color colour) {
+        // Find coordinate of shadow in Haversine format
+        final Location shadowCoordinates = new Location(
+                shadowLocation.getLat() - Distance.Haversine.kilometersToDecimalDegrees(shadowRadius / 1000),
+                shadowLocation.getLon() - Distance.Haversine.kilometersToDecimalDegrees(shadowRadius / 1000));
+
+        // Get all the locations of the edges of the shadow
+        final List<Location> shadowLocations = generateCircle(shadowLocation, shadowCoordinates);
+
+        // Project the circular show onto the world
+        projectShadowCoordinates(shadowLocations);
+
+        // Get the actual shadow svg path
+        final String shadowPath = generatePath(shadowLocations);
+
+        // Add that shadow path to the graphics group
+        addShadowToGroup(shadowPath, colour);
+    }
+
+    /**
+     * Adds shadow graphic to group
+     *
+     * @param shadowPath - the raw string svg path
+     * @param colour
+     */
+    private void addShadowToGroup(final String shadowPath, final Color colour) {
+        // Create the shadow graphic and colour it
+        final SVGPath shadowMarker = new SVGPath();
+        shadowMarker.setContent(shadowPath);
+        shadowMarker.setFill(colour);
+
+        // Get rid of borders and allow events to pass through the shadow
+        shadowMarker.setStrokeWidth(0);
+        shadowMarker.setMouseTransparent(true);
+
+        dayNightGroup.getChildren().add(shadowMarker);
+    }
+
+    /**
+     * Convert lat and long coordinates of shadow edges to x and y coords
+     *
+     * @param locations - array of Location objects that only have their
+     * lattitude and longitude fields calculated
+     */
+    private void projectShadowCoordinates(final List<Location> locations) {
+        // For every location calculate its x and y coordinate
+        locations.forEach(location -> {
+
+            location.setX(MarkerUtilities.longToX(location.getX(), MapView.MIN_LONG, MapView.MAP_WIDTH, MapView.MAX_LONG - MapView.MIN_LONG));
+            location.setY(MarkerUtilities.latToY(location.getY(), MapView.MAP_WIDTH, MapView.MAP_HEIGHT) - SHADOW_LOCATION_Y_OFFSET);
+
+
+        });
+
+    }
+
+    /**
+     * Generate the actual path from the shadow locations
+     *
+     * @param locations
+     * @return the string that containing the raw svg path
+     */
+    private String generatePath(final List<Location> locations) {
+        String path = "";
+        boolean first = true;
+
+        // Loop through all the locations
+        for (int i = 0; i < locations.size(); ++i) {
+            if (Double.isNaN(locations.get(i).getY()) || Double.isNaN(locations.get(i).getX())) {
```

Comment:
just noting that subsequent calls in for loop frequently call `locations.get(i)`, might it make sense to extract it into a variable (maybe one step further and having variables for x and y)